src/DragAndDrop/Draggable/components/DraggableSource.js

```diff
@@ -94,17 +133,25 @@ const source = {
   },
   isDragging: ({ id, containerId, groupName }, monitor) => {
     const item = monitor.getItem();
-    const isSameGroup =
-      groupName && item.groupName && groupName === item.groupName;
     const isSameContainer = containerId === item.containerId;
-    return (isSameGroup || isSameContainer) && item.id === id;
+
+    // tracking draggable item`
+    dragObject.originalItem = item.originalItem;
+    dragObject.addedToContainerId = containerId;
+    dragObject.removedFromContainerId = item.containerId;
+    dragObject.removedIndex = item.originalIndex;
+    dragObject.groupName = groupName;
+
+    return (isSameGroup(item, groupName) || isSameContainer) && item.id === id;
   },
 };
 
 const collect = (connect, monitor) => ({
   connectDragSource: connect.dragSource(),
   connectDragPreview: connect.dragPreview(),
   isDragging: monitor.isDragging(),
+  // Need to set added index from component, since isDragging index is one step behind
+  setAddedIndex: index => (dragObject.addedIndex = index),
```

Comment:
```suggestion
  setAddedIndex: index => { dragObject.addedIndex = index },
```

let's do regular code block curly brackets, braces might trip over other readers